src/dyn/Energy_and_Forces.cpp

```diff
@@ -315,7 +319,9 @@ void update_forces(dyn_control_params& prms, dyn_variables& dyn_vars, nHamiltoni
 
     // TSH or adiabatic (including excited states)
     // state-specific forces   
-    vector<int> effective_states(dyn_vars.act_states);
+    vector<int> effective_states(ntraj, 0);
+    if(prms.rep_sh==1){ effective_states = dyn_vars.act_states; }
+    else{ effective_states = dyn_vars.act_states_dia; }

```

Comment:
here and in other similar situations, please use if/else if  construction, instead of if/else